packages/openid4vc/src/shared/router/context.ts

```diff
@@ -83,7 +83,7 @@ export function sendErrorResponse(
   error: unknown,
   additionalPayload?: Record<string, unknown>
 ) {
-  const body = { error: message, ...additionalPayload }
+  const body = { error: message, ...(error instanceof Error && { cause: error.message }), ...additionalPayload }
```

Comment:
This is not compliant with the spec. there is error and error_description. But we should be cautious with just returning the error message for unknown errors.

For oid4vci i reworked it to now throw a `Oauth2ServerErrorResponseError ` and in the place where `sendErrorResponse` is called i first check for   instanceof Oauth2ServerErrorResponseError and then call `sendOauth2ErrorResponse`. Then we make sure only error messages that should be leaked to the client are leaked. (Because we only construct a `Oauth2ServerErrorResponseError` if needed. Can you udpate?